### PR TITLE
stmt-reference, sysvars: Document dml-batch-size

### PR DESCRIPTION
**Pull request by ireneontheway**
### What is changed, added or deleted? (Required)

Part of https://github.com/pingcap/docs/issues/3155

Only relevant to master.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/3838 https://github.com/pingcap/docs/pull/3977
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: sql-statements/sql-statement-load-data.md [hunk 1]
```diff
@@ -97,13 +97,16 @@ LOAD DATA LOCAL INFILE '/mnt/evo970/data-sets/bikeshare-data/2017Q4-capitalbikes
 
 ## MySQL 兼容性
 
-* 默认情况下，TiDB 每 20,000 行会进行一次提交。这类似于 MySQL NDB Cluster，但并非 InnoDB 存储引擎的默认配置。
+`LOAD DATA` 语句应该完全兼容 MySQL。所有兼容性差异都会汇总于 GitHub 的 [issue](https://github.com/pingcap/tidb/issues/new/choose) 中。
```
Comment (TomShawn) [inline sql-statements/sql-statement-load-data.md:100]:
```suggestion
`LOAD DATA` 语句应该完全兼容 MySQL。若发现任何兼容性差异，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues/new/choose)。
```

File: sql-statements/sql-statement-load-data.md [hunk 2]
```diff
@@ -97,13 +97,16 @@ LOAD DATA LOCAL INFILE '/mnt/evo970/data-sets/bikeshare-data/2017Q4-capitalbikes
...
 
 > **注意：**
 >
-> 这种拆分事务提交的方式是以打破事务的原子性和隔离性为代价的，使用该特性时，使用者需要保证没有其他对正在处理的表的**任何**操作，并且在出现报错时，需要及时**人工介入，检查数据的一致性和完整性**。因此，不建议对读写频繁的表使用 `LOAD DATA` 语句。
+> 在 TiDB 的早期版本中，`LOAD DATA` 语句每 20000 行提交一次。新版本的 TiDB 默认在一个事务中提交所有行。从 TiDB 4.0 及以前版本升级后，可能出现 `ERROR 8004 (HY000) at line 1: Transaction is too large, size: 100000058` 错误。
+>
+> 建议增加 `tidb.toml` 文件中的 `txn-total-size-limit` 值来解决这一问题。如果无法增加此限制，还可以将 [`tidb_dml_batch_size`](/system-variables.md#tidb_dml_batch_size) 的值设置为 20000 来恢复以前的行为。
```
Comment (TomShawn) [inline sql-statements/sql-statement-load-data.md:106]:
```suggestion
> 要解决该问题，建议调大 `tidb.toml` 文件中的 `txn-total-size-limit` 值。如果无法增加此限制，还可以将 [`tidb_dml_batch_size`](/system-variables.md#tidb_dml_batch_size) 的值设置为 `20000` 来恢复升级前的行为。
```

File: system-variables.md [hunk 3]
```diff
@@ -304,6 +304,14 @@ SET  GLOBAL tidb_distsql_scan_concurrency = 10;
 - 这个变量用来设置 scan 操作的并发度。
 - AP 类应用适合较大的值，TP 类应用适合较小的值。对于 AP 类应用，最大值建议不要超过所有 TiKV 节点的 CPU 核数。
 
+### tidb_dml_batch_size
```
Comment (TomShawn) [inline system-variables.md:307]:
```suggestion
### `tidb_dml_batch_size`
```

Comment (ireneontheway) [reply system-variables.md]: Done in https://github.com/pingcap/docs-cn/pull/4590/commits/5a0f3aecb7290361496a61b00527f3de4c2e7672

File: system-variables.md [hunk 4]
```diff
@@ -304,6 +304,14 @@ SET  GLOBAL tidb_distsql_scan_concurrency = 10;
 - 这个变量用来设置 scan 操作的并发度。
 - AP 类应用适合较大的值，TP 类应用适合较小的值。对于 AP 类应用，最大值建议不要超过所有 TiKV 节点的 CPU 核数。
 
+### `tidb_dml_batch_size`
+
+- 作用域：SESSION
+- 默认值：0
+- 样本值：20000
+- 这个变量的值大于 `0` 时，TiDB 会分批提交 `INSERT` 或 `LOAD DATA` 等语句为更小的事务。这样减少了内存的使用，有助于确保批量修改不会涉及 `txn-total-size-limit`。
```
Comment (TomShawn) [inline system-variables.md:312]: @jackysp PTAL whether my comment is technically correct, thanks!
```suggestion
- 这个变量的值大于 `0` 时，TiDB 会将 `INSERT` 或 `LOAD DATA` 等语句在更小的事务中批量提交。这样可减少内存使用，确保大批量修改时事务大小不会达到 `txn-total-size-limit` 限制。
```

File: system-variables.md [hunk 5]
```diff
@@ -304,6 +304,14 @@ SET  GLOBAL tidb_distsql_scan_concurrency = 10;
...
+- 只有变量值为 `0` 时才符合 ACID 特性。变量值不为 `0` 时，无法保证 TiDB 的原子性和隔离性。
```
Comment (TomShawn) [inline system-variables.md:313]:
```suggestion
- 只有变量值为 `0` 时才符合 ACID 要求。否则无法保证 TiDB 的原子性和隔离性要求。
```

**Comment by cfzjywxk** [comment]
LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-load-data.md [hunk 6]
```diff
@@ -97,13 +97,16 @@ LOAD DATA LOCAL INFILE '/mnt/evo970/data-sets/bikeshare-data/2017Q4-capitalbikes
 
 ## MySQL 兼容性
 
-* 默认情况下，TiDB 每 20,000 行会进行一次提交。这类似于 MySQL NDB Cluster，但并非 InnoDB 存储引擎的默认配置。
+`LOAD DATA` 语句应该完全兼容 MySQL。若发现任何兼容性差异，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues/new/choose)。
 
 > **注意：**
 >
-> 这种拆分事务提交的方式是以打破事务的原子性和隔离性为代价的，使用该特性时，使用者需要保证没有其他对正在处理的表的**任何**操作，并且在出现报错时，需要及时**人工介入，检查数据的一致性和完整性**。因此，不建议对读写频繁的表使用 `LOAD DATA` 语句。
+> 在 TiDB 的早期版本中，`LOAD DATA` 语句每 20000 行提交一次。新版本的 TiDB 默认在一个事务中提交所有行。从 TiDB 4.0 及以前版本升级后，可能出现 `ERROR 8004 (HY000) at line 1: Transaction is too large, size: 100000058` 错误。
```
Comment (TomShawn) [inline sql-statements/sql-statement-load-data.md:104]:
```suggestion
> 在 TiDB 的早期版本中，`LOAD DATA` 语句每 20000 行进行一次提交。新版本的 TiDB 默认在一个事务中提交所有行。从 TiDB 4.0 及以前版本升级后，可能出现 `ERROR 8004 (HY000) at line 1: Transaction is too large, size: 100000058` 错误。
```

**Pull request merged**